### ISSUE TITLE
Unexpected NaN gradient when raising `inf + 0j` to 1

### ISSUE THREAD

**Issue by lfaucheux** [google/jax#18749]
### Description

I am getting unexpected nan gradient with the following example.

```python
>>> import jax; jax.config.update("jax_enable_x64", True)
>>> (g := jax.grad(lambda x1: jax.numpy.power(x1, 1.).real))(float('inf') + 0j)
Array(nan+nanj, dtype=complex128)
# >>> g(float('inf'))
# Array(1., dtype=float64, weak_type=True)
```

### What jax/jaxlib version are you using?

jax v0.4.20, jaxlib v0.4.20 

### Which accelerator(s) are you using?

CPU | GPU

### Additional system info?

Python 3.10, Under Linux | Windows

**Comment by jakevdp** [comment]
Thanks for the report! The root cause of this issue seems to be this:
```python
>>> print(jax.lax.pow(complex('inf'), 0))
(nan+nanj)
```
I'll note, however, that if you use integer rather than floating point powers in `jax.numpy.power`, then it lowers to `lax.integer_pow` and returns the expected answer:
```python
>>> import jax
>>> print(jax.grad(lambda x1: jax.numpy.power(x1, 1).real)(float('inf') + 0j))
# (1+0j)
```
Since `lax.pow` basically dispatches directly to XLA:Pow, I think this should probably be reported at https://github.com/openxla/xla.

**Comment by jakevdp** [comment]
Reported in https://github.com/openxla/xla/issues/7440

**Comment by jakevdp** [comment]
It looks like this is fixed now. Here's the result in jax 0.4.26:
```python
>>> import jax; jax.config.update("jax_enable_x64", True)
>>> (g := jax.grad(lambda x1: jax.numpy.power(x1, 1.).real))(float('inf') + 0j)
Array(1.+0.j, dtype=complex128)
```
